code/modules/projectiles/ammunition.dm

```diff
@@ -311,34 +311,36 @@
 
 /obj/item/ammo_magazine/resolve_attackby(atom/A, mob/user)
 	//Clicking on tile with no collectible items will empty it, if it has the verb to do that.
-	if(isturf(A))
-		src.quick_empty(A)
+	if(isturf(A) && !A.density)
+		dump_it(A)
 		return TRUE
 	return ..()
 
-/obj/item/ammo_magazine/verb/quick_empty(var/turf/target)
+/obj/item/ammo_magazine/verb/quick_empty()
 	set name = "Empty Ammo Container"
 	set category = "Object"
+	set src in view(1)
 
 	if((!ishuman(usr) && (src.loc != usr)) || usr.stat || usr.restrained())
 		return
 
-	var/turf/T
-	if(isturf(target))
-		T = target
-	else
-		T = get_turf(src)
-
+	var/turf/T = get_turf(src)
 	if(!istype(T))
 		return
+	dump_it(T, usr)
```

Comment:
That call didn't match proc interface (line 332) `proc/dump_it(var/turf/target`)